### ISSUE TITLE
[dev.icinga.com #4197] Add support for LDAP users with multiple uid's

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga-web#1143]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/4197

**Created by ryanlee on 2013-05-17 18:58:19 +00:00**

Assignee: _mfrosch_
Status: _Resolved_  (closed on _2013-10-10 12:28:58 +00:00_)
Target Version: _1.10_
Last Update: _2013-10-16 14:46:31 +00:00 (in Redmine)_

---

The LDAP schema allows for multiple uid attributes per user, but the code for getLdaprecord in /usr/share/icinga-web/app/modules/AppKit/models/Auth/Provider/LDAPModel.class.php does not appear to take this into account, throwing this error in the debug log and preventing login from succeeding:

Uncaught AppKitPHPError: PHP Error Undefined index: uid (/usr/share/icinga-web/app/modules/AppKit/models/Auth/Provider/LDAPModel.class.php:88) (/usr/share/icinga-web/app/modules/AppKit/lib/logging/AppKitExceptionHandler.class.php:52)

**Changesets**

_2013-10-07 15:57:25 +00:00_ by _mfrosch_ ab7f9ca00eacb1965a3e6826242b1a531ad47118

    Removed ldap_userattr matching to allow multiple UIDs of a user
    
    Basically are we ignoring the userattribute now and plainly depend on the
    search filter to match the correct user.
    
    Additionally requiring the token __USERNAME__ in the ldap_filter_user setting.
    
    refs #4197
